src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntryValidator.cs

```diff
@@ -90,23 +88,35 @@ private void ValidateCustomTagEntry(CustomTagEntry tagEntry)
             if (QueryLimit.AllInstancesTags.Contains(tag))
             {
                 throw new CustomTagEntryValidationException(
-                   string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidCustomTag, tag.DictionaryEntry.Name));
+                   string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidCustomTag, tagEntry.Path));
             }
 
             if (tag.IsPrivate)
             {
+                if (string.IsNullOrWhiteSpace(tagEntry.PrivateCreator))
```

Comment:
can you move this logic to a function ValidatePrivateCreator(Tag) . In there you may have to do length check along with current null or not null check

Reply:
All LO VR checks

Reply:
will fix it.